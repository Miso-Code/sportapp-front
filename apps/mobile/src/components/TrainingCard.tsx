import React from 'react'
import dayjs from 'dayjs'
import { StyleSheet } from 'react-native'
import { Card, Text, Chip, useTheme, MD3Theme } from 'react-native-paper'
import { useTranslation } from 'react-i18next'

interface TrainingCardProps {
	date: Date
	title: string
	description: string
}

const TrianingCard: React.FC<TrainingCardProps> = ({
	date,
	title,
	description
}) => {
	const theme = useTheme()
	const styles = createStyles(theme)
	const { t, i18n } = useTranslation()

	const parsedDate = dayjs(date).locale(i18n.language).format('DD MMMM YYYY')

	const isUpcoming = dayjs(date).isAfter(dayjs(), 'days')

	return (
		<Card elevation={1} style={styles.card}>
			<Card.Content>
				<Text variant='labelLarge' style={styles.dateLabel}>
					{parsedDate}
				</Text>
				<Text variant='titleLarge' style={styles.cardTitle}>
					{title}
				</Text>
				<Text variant='bodyMedium'>
					{description ||
						'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.'}
				</Text>
				<Chip
					style={isUpcoming ? styles.chipUpcoming : styles.chipToday}
					textStyle={
						isUpcoming
							? styles.chipUpcomingText
							: styles.chipTodayText
					}
					compact>
					{t(isUpcoming ? 'training.upcoming' : 'training.today')}
				</Chip>
			</Card.Content>
		</Card>
	)
}

const createStyles = (theme: MD3Theme) =>
	StyleSheet.create({
		card: {
			backgroundColor: '#f2f2f2',
			paddingBottom: 10
		},
		cardTitle: {
			fontWeight: 'bold'
		},
		dateLabel: {
			opacity: 0.5
		},
		chipToday: {
			right: 10,
			top: 10,
			position: 'absolute',
			borderWidth: 1,
			borderColor: theme.colors.primary,
			backgroundColor: theme.colors.primaryContainer,
			borderRadius: 50
		},

		chipUpcoming: {
			right: 10,
			top: 10,
			position: 'absolute',
			borderWidth: 1,
			borderColor: theme.colors.error,
			backgroundColor: theme.colors.errorContainer,
			borderRadius: 50
		},
		chipTodayText: {
			color: theme.colors.primary
		},
		chipUpcomingText: {
			color: theme.colors.error
		}
	})

export default TrianingCard
