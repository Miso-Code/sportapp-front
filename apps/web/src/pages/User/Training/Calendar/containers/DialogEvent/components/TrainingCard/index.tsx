import {
	Card,
	CardContent,
	List,
	ListItem,
	Typography
} from '@mui/material'
import { format } from 'date-fns'
import { es } from 'date-fns/locale'
import { useTranslation } from 'react-i18next'
import { CustomEvent } from '../../../CustomCalendar/interfaces'

export default function SessionCard({
	selectedValue
}: {
	selectedValue: CustomEvent
}) {
	const { t } = useTranslation()
	return (
		<Card>
			<CardContent>
				<Typography variant='body1' color='GrayText'>
					{format(
						selectedValue.start as Date,
						'dd MMMM yyyy hh:mm a',
						{
							locale: es
						}
					)}
				</Typography>
				<Typography gutterBottom variant='h6'>
					{selectedValue.title}
				</Typography>
				<List disablePadding>
					<ListItem
						disablePadding
						sx={{
							display: 'flex',
							justifyContent: 'space-between'
						}}>
						<Typography
							sx={{
								fontWeight: 700
							}}>
							{t('training.warmUp')}
						</Typography>
						<Typography>
							{(selectedValue.warm_up || 0) * 60}
						</Typography>
					</ListItem>
					<ListItem
						disablePadding
						sx={{
							display: 'flex',
							justifyContent: 'space-between'
						}}>
						<Typography
							sx={{
								fontWeight: 700
							}}>
							{t('training.cardio')}
						</Typography>
						<Typography>
							{(selectedValue.cardio || 0) * 60}
						</Typography>
					</ListItem>
					<ListItem
						disablePadding
						sx={{
							display: 'flex',
							justifyContent: 'space-between'
						}}>
						<Typography
							sx={{
								fontWeight: 700
							}}>
							{t('training.strength')}
						</Typography>
						<Typography>
							{(selectedValue.strength || 0) * 60}
						</Typography>
					</ListItem>
					<ListItem
						disablePadding
						sx={{
							display: 'flex',
							justifyContent: 'space-between'
						}}>
						<Typography
							sx={{
								fontWeight: 700
							}}>
							{t('training.coolDown')}
						</Typography>
						<Typography>
							{(selectedValue.cool_down || 0) * 60}
						</Typography>
					</ListItem>
				</List>
			</CardContent>
		</Card>
	)
}
