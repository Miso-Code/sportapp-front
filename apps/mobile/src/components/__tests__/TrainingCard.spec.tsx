import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'

import TrainingCard from '../TrainingCard'

jest.mock('dayjs')

describe('TrainingCard', () => {
	let component: ReactTestRenderer

	beforeEach(() => {
		component = renderer.create(
			<TrainingCard
				date={new Date()}
				trainingSession={{
					training_plan_session_id: '1',
					weekday: 'monday',
					start_time: '10:00 AM',
					warm_up: 0.35,
					cardio: 0.7,
					strength: 0.525,
					cool_down: 0.175,
					user_id: '1'
				}}
			/>
		)
	})

	afterEach(() => {
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render today chip', () => {
		expect(
			component.root.findByProps({ testID: 'chip' }).props.children
		).toBe('training.today')
	})

	it('should change the chip on a future date', () => {
		const future = new Date()
		future.setFullYear(future.getFullYear() + 1)
		component.update(
			<TrainingCard
				trainingSession={{
					training_plan_session_id: '1',
					weekday: 'monday',
					start_time: '10:00 AM',
					warm_up: 0.35,
					cardio: 0.7,
					strength: 0.525,
					cool_down: 0.175,
					user_id: '1'
				}}
				date={future}
			/>
		)

		expect(
			component.root.findByProps({ testID: 'chip' }).props.children
		).toBe('training.upcoming')
	})
})
