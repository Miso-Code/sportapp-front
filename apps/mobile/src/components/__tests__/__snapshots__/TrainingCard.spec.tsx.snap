// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingCard should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#f2f2f2",
      "borderRadius": 12,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f2f2f2",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "paddingBottom": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    >
      <View
        borderRadiusStyles={{}}
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            {
              "paddingVertical": 16,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          2024-05-04
        </Text>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          10:00 AM
        </Text>
        <View
          style={
            {
              "marginVertical": 10,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              training.warmUp
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              21
               
              training.minutes
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              training.cardio
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              42
               
              training.minutes
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              training.strength
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              32
               
              training.minutes
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              training.coolDown
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              11
               
              training.minutes
            </Text>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "500",
                      "letterSpacing": 0.15,
                      "lineHeight": 24,
                    },
                    undefined,
                  ],
                ]
              }
            >
              training.totalDuration
            </Text>
            <Text
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "letterSpacing": 0.15,
                      "lineHeight": 24,
                    },
                    undefined,
                  ],
                ]
              }
            >
              1.8
               
              training.hours
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 50,
              "position": "absolute",
              "right": 10,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "top": 10,
            }
          }
          testID="chip-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(234, 221, 255, 1)",
                "borderColor": "rgba(103, 80, 164, 1)",
                "borderRadius": 50,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flex": undefined,
                "flexDirection": "column",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": true,
                  "expanded": undefined,
                  "selected": false,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  [
                    {
                      "borderRadius": 50,
                    },
                    {
                      "width": "100%",
                    },
                  ],
                ]
              }
              testID="chip"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "paddingLeft": 4,
                      "position": "relative",
                    },
                    {
                      "paddingLeft": 0,
                    },
                    {
                      "paddingRight": 0,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        [
                          {
                            "marginVertical": 6,
                            "textAlignVertical": "center",
                          },
                          {
                            "color": "rgba(29, 25, 43, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          {
                            "marginLeft": 12,
                            "marginRight": 12,
                          },
                          {
                            "color": "rgba(103, 80, 164, 1)",
                          },
                        ],
                      ],
                    ]
                  }
                >
                  training.today
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
